src/api/posts.ts

```diff
@@ -0,0 +1,25 @@
+import { Post } from '../react-app-env';
+
+export const BASE_URL = 'https://mate.academy/students-api/';
+
+export async function getPosts(userId: string): Promise<Post[]> {
+  const response = await fetch(`${BASE_URL}/posts?userId=${userId}`);
+
+  if (!response.ok) {
+    throw new Error(`${response.status} - ${response.statusText}`);
+  }
+
+  return response.json();
+}
+
+export async function getPostbyId(postId:number) {
```

Comment:
Missed requirement: Create `getPostDetails(postId)` method in `/src/api/posts.ts` and load post details from `/posts/:postId`.
```suggestion
export async function getPostDetails(postId:number) {
```